packages/router/router.ts

```diff
@@ -2123,12 +2119,6 @@ export function createRouter(init: RouterInit): Router {
 
     if (blockerFunctions.get(key) !== fn) {
       blockerFunctions.set(key, fn);
-      if (activeBlocker == null) {
-        // This is now the active blocker
-        activeBlocker = key;
-      } else if (key !== activeBlocker) {
-        warning(false, "A router only supports one blocker at a time");
-      }
```

Comment:
While testing, I realized that this `activeBlocker` key storage was broken in strict mode since the effect cleanup runs _after_ the second render.  So we would get the following:

```
getBlocker('singleton-key');
getBlocker('singleton-key');
deletBlocker('singleton-key');
```

Causing us to think we did not have a blocker.  But that would get resolved as soon as we entered any form data causing the getBlocker to run again on the `fn` identity change.  See comments below for the new Map-only approach